File: src/main/java/wraith/fabricaeexnihilo/modules/tools/CrookTool.java
package wraith.fabricaeexnihilo.modules.tools;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.block.BlockState;
import net.minecraft.item.ItemStack;
import net.minecraft.item.ToolMaterial;
import wraith.fabricaeexnihilo.api.registry.FabricaeExNihiloRegistries;
import wraith.fabricaeexnihilo.modules.ModTags;

public class CrookTool extends ToolItemWithRegistry {

    public CrookTool(ToolMaterial material, FabricItemSettings settings) {
        super(material, FabricaeExNihiloRegistries.CROOK, settings);
    }

    @Override
    public boolean isSuitableFor(BlockState state) {
        return FabricaeExNihiloRegistries.CROOK.isRegistered(state.getBlock());
    }

    public static boolean isCrook(ItemStack stack) {
        return stack.getItem() instanceof CrookTool || ModTags.CROOK_TAG.contains(stack.getItem());
    }

}

File: src/main/kotlin/exnihilofabrico/modules/tools/CrookTool.kt
package exnihilofabrico.modules.tools

import exnihilofabrico.api.registry.ExNihiloRegistries
import exnihilofabrico.modules.ModTags.CROOK_TAG
import net.minecraft.block.BlockState
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.item.ToolMaterial

class CrookTool(material: ToolMaterial, settings: Item.Settings):
    ToolItemWithRegistry(material, ExNihiloRegistries.CROOK, settings) {

    override fun isEffectiveOn(state: BlockState) = ExNihiloRegistries.CROOK.isRegistered(state.block)

    companion object {
        @JvmStatic
        fun isCrook(stack: ItemStack): Boolean {
            if(stack.item is CrookTool || CROOK_TAG.contains(stack.item))
                return true
            return false
        }
    }

}